src/Lecture2.hs

```diff
@@ -70,7 +87,8 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt idx xs = (xs !? idx, take idx xs ++ drop (idx + 1) xs)
```

Comment:
That's a correct implementation 👍🏻 
Alternatively, you can use a standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of traversing the list prefix three times